src/Servers/HttpSys/src/MessagePump.cs

```diff
@@ -301,28 +238,142 @@ public void Dispose()
             Listener.Dispose();
         }
 
-        private class ApplicationWrapper<TContext> : IHttpApplication<object>
+        private interface IWorkerFactory
         {
-            private readonly IHttpApplication<TContext> _application;
+            IThreadPoolWorkItem Get(HttpSysListener server, NativeRequestContext requestContext);
+        }
+
+        private sealed class WorkerFactory<TContext> : IWorkerFactory
+        {
+            private const int _maxPooledContexts = 512;
```

Comment:
Picking 512 for the _maxPooledContexts seems pretty arbitrary. I feel that it's tuned to perform well in benchmarks that will never establish more that 512 concurrent connections.

At the same time, it seems way too implementation-specific to be adding a knob for this.

Reply:
BTW I'm not suggesting we change this necessarily. I just wanted to bring it up so everyone reviewing the PR can consider it.